lib/taurus/qt/qtgui/taurusgui/config_loader/abstract.py

```diff
@@ -0,0 +1,180 @@
+#############################################################################
+##
+# This file is part of Taurus
+##
+# http://taurus-scada.org
+##
+# Copyright 2011 CELLS / ALBA Synchrotron, Bellaterra, Spain
+##
+# Taurus is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+##
+# Taurus is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+##
+# You should have received a copy of the GNU Lesser General Public License
+# along with Taurus.  If not, see <http://www.gnu.org/licenses/>.
+##
+###########################################################################
+
+""""""
+import abc
+
+# Python 2/3 compatibility
+if not hasattr(abc, "ABC"):
+    setattr(abc, "ABC", abc.ABCMeta("ABC", (object,), {}))
+
+
+__all__ = ["AbstractConfigLoader", "ConfigLoaderError"]
+
+
+class ConfigLoaderError(Exception):
+    """
+    Base exception raised by ConfigLoader
+    """
+
+    def __init__(self, message):
+        message = "Exception raised while loading configuration: " + message
+        super(ConfigLoaderError, self).__init__(message)
+
+
+class AbstractConfigLoader(abc.ABC):
+    """
+    Abstract class for config loaders.
+    It defines interface which has to be implemneted by subclass in
+    order for it to be ConfigLoader
+    """
+
+    def __init__(self, confname):
+        self._confname = confname
+
+    @abc.abstractmethod
+    def supports(self, confname):
+        """
+        Return True or False for support of specific configuration passed
+        """
+        return None
+
+    @abc.abstractmethod
+    def load(self):
+        """
+        This method is meant to load actual data from file on disk
+        """
+
+    @abc.abstractproperty
+    def conf_dir(self):
```

Comment:
Is this really necessary? It constrains the loaders to be file-based. I think that with a bit of care it should be easy to leave the possibility open to e.g. database-based config loaders

Reply:
I've added it since some value was required to set for `TaurusGui._confDirectory`. However, I think it can be removed from abstract interface and just fallback to `$HOME` or `$HOME/.config/<ORGANIZATION>`